backend/lib/services/cloudProviderCredentials.js

```diff
@@ -0,0 +1,153 @@
+//
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+const _ = require('lodash')
+const { getQuotas } = require('../cache')
+const { Resources } = require('@gardener-dashboard/kube-client')
+const { encodeBase64 } = require('../utils')
+const createError = require('http-errors')
+const logger = require('../logger')
+
+exports.list = async function ({ user, body }) {
+  const client = user.client
+  const { coordinate: { namespace } } = body
+
+  const [
+    { items: secretBindings },
+    { items: referencedSecrets },
+  ] = await Promise.all([
+    client['core.gardener.cloud'].secretbindings.list(namespace),
+    client.core.secrets.list(namespace, { labelSelector: 'reference.gardener.cloud/secretbinding=true' }),
+  ])
+
+  const credentialsList = secretBindings.map(secretBinding => {
+    const secret = referencedSecrets.find(secret => secret.metadata.name === secretBinding.secretRef.name)
+
+    return {
+      secretBinding,
+      secret,
+      quotas: resolveQuotas(secretBinding),
+    }
+  })
+
+  return credentialsList
+}
+
+exports.create = async function ({ user, body }) {
+  const client = user.client
+  const { coordinate: { namespace, name }, credential: { poviderType, secretData } } = body
```

Comment:
I would not use the property `coordinate`. Use the relevant properties flat. Pass the params and not body. 
```
const { namespace, name, poviderType, secretData } = params
```

Reply:
Update: I talked to @petersutter and we think it should be:
```
const { secretBindingNamespace, secretBindingName, poviderType, secretData } = params
```
This described that fact that the primary usage for name and namespace is the secretBinding. We could als think of of:
```
const { namespace, name, secretBindingName, secretName = secretBindingName, poviderType, secretData } = params
```